### PR TITLE
Rename "Orion"

### PR DESCRIPTION
**Pull request by madkinsz**
## Pull requests

- https://github.com/PrefectHQ/prefect/pull/8434
- https://github.com/PrefectHQ/prefect/pull/8460
- https://github.com/PrefectHQ/prefect/pull/8461
- https://github.com/PrefectHQ/prefect/pull/8463
- https://github.com/PrefectHQ/prefect/pull/8464
- https://github.com/PrefectHQ/prefect/pull/8482
- https://github.com/PrefectHQ/prefect/pull/8483
- https://github.com/PrefectHQ/prefect/pull/8482
- #8469 
- #8486 
- #8487 
- #8490 
- #8488 
- #8489 
- https://github.com/PrefectHQ/prefect/pull/8492
- #8491 
- #8499
- #8496
- #8500 

## Changes

- Default SQLite database name changed from `orion.db` to `prefect.db`
- Logger `prefect.orion` renamed to `prefect.server`
- Constant `ORION_API_VERSION` renamed to `SERVER_API_VERSION`
- Kubernetes deployment template application name changed from `prefect-orion` to `prefect-server`
- Command `prefect kubernetes manifest orion` renamed to `prefect kubernetes manifest server`
- Log config  handler `orion` renamed to `api`
- Class `OrionLogWorker` renamed to `APILogWorker`
- Class `OrionHandler` renamed to `APILogHandler`
- Directory `orion-ui` renamed to `ui`
- Class `OrionRouter` renamed to `PrefectRouter`
- Class `OrionAPIRoute` renamed to `PrefectAPIRoute`
- Class `OrionDBInterface` renamed to `PrefectDBInterface`
- Class `OrionClient` renamed to `PrefectClient`
- Module `prefect.client.orion` renamed to `prefect.client.orchestration`
- Command group `prefect orion` renamed to `prefect server`
- Module `prefect.orion` renamed to `prefect.server`
- The following settings have been renamed:
    - `PREFECT_LOGGING_ORION_ENABLED` → `PREFECT_LOGGING_TO_API_ENABLED`
    - `PREFECT_LOGGING_ORION_BATCH_INTERVAL` → `PREFECT_LOGGING_TO_API_BATCH_INTERVAL`
    - `PREFECT_LOGGING_ORION_BATCH_SIZE` → `PREFECT_LOGGING_TO_API_BATCH_SIZE`
    - `PREFECT_LOGGING_ORION_MAX_LOG_SIZE` → `PREFECT_LOGGING_TO_API_MAX_LOG_SIZE`
    - `PREFECT_LOGGING_ORION_WHEN_MISSING_FLOW` → `PREFECT_LOGGING_TO_API_WHEN_MISSING_FLOW`
    - `PREFECT_ORION_BLOCKS_REGISTER_ON_START` → `PREFECT_API_BLOCKS_REGISTER_ON_START`
    - `PREFECT_ORION_DATABASE_CONNECTION_URL` → `PREFECT_API_DATABASE_CONNECTION_URL`
    - `PREFECT_ORION_DATABASE_MIGRATE_ON_START` → `PREFECT_API_DATABASE_MIGRATE_ON_START`
    - `PREFECT_ORION_DATABASE_TIMEOUT` → `PREFECT_API_DATABASE_TIMEOUT`
    - `PREFECT_ORION_DATABASE_CONNECTION_TIMEOUT` → `PREFECT_API_DATABASE_CONNECTION_TIMEOUT`
    - `PREFECT_ORION_SERVICES_SCHEDULER_LOOP_SECONDS` → `PREFECT_API_SERVICES_SCHEDULER_LOOP_SECONDS`
    - `PREFECT_ORION_SERVICES_SCHEDULER_DEPLOYMENT_BATCH_SIZE` → `PREFECT_API_SERVICES_SCHEDULER_DEPLOYMENT_BATCH_SIZE`
    - `PREFECT_ORION_SERVICES_SCHEDULER_MAX_RUNS` → `PREFECT_API_SERVICES_SCHEDULER_MAX_RUNS`
    - `PREFECT_ORION_SERVICES_SCHEDULER_MIN_RUNS` → `PREFECT_API_SERVICES_SCHEDULER_MIN_RUNS`
    - `PREFECT_ORION_SERVICES_SCHEDULER_MAX_SCHEDULED_TIME` → `PREFECT_API_SERVICES_SCHEDULER_MAX_SCHEDULED_TIME`
    - `PREFECT_ORION_SERVICES_SCHEDULER_MIN_SCHEDULED_TIME` → `PREFECT_API_SERVICES_SCHEDULER_MIN_SCHEDULED_TIME`
    - `PREFECT_ORION_SERVICES_SCHEDULER_INSERT_BATCH_SIZE` → `PREFECT_API_SERVICES_SCHEDULER_INSERT_BATCH_SIZE`
    - `PREFECT_ORION_SERVICES_LATE_RUNS_LOOP_SECONDS` → `PREFECT_API_SERVICES_LATE_RUNS_LOOP_SECONDS`
    - `PREFECT_ORION_SERVICES_LATE_RUNS_AFTER_SECONDS` → `PREFECT_API_SERVICES_LATE_RUNS_AFTER_SECONDS`
    - `PREFECT_ORION_SERVICES_PAUSE_EXPIRATIONS_LOOP_SECONDS` → `PREFECT_API_SERVICES_PAUSE_EXPIRATIONS_LOOP_SECONDS`
    - `PREFECT_ORION_SERVICES_CANCELLATION_CLEANUP_LOOP_SECONDS` → `PREFECT_API_SERVICES_CANCELLATION_CLEANUP_LOOP_SECONDS`
    - `PREFECT_ORION_API_DEFAULT_LIMIT` → `PREFECT_API_DEFAULT_LIMIT`
    - `PREFECT_ORION_API_HOST` → `PREFECT_SERVER_API_HOST`
    - `PREFECT_ORION_API_PORT` → `PREFECT_SERVER_API_PORT`
    - `PREFECT_ORION_API_KEEPALIVE_TIMEOUT` → `PREFECT_SERVER_API_KEEPALIVE_TIMEOUT`
    - `PREFECT_ORION_UI_ENABLED` → `PREFECT_UI_ENABLED`
    - `PREFECT_ORION_UI_API_URL` → `PREFECT_UI_API_URL`
    - `PREFECT_ORION_ANALYTICS_ENABLED` → `PREFECT_SERVER_ANALYTICS_ENABLED`
    - `PREFECT_ORION_SERVICES_SCHEDULER_ENABLED` → `PREFECT_API_SERVICES_SCHEDULER_ENABLED`
    - `PREFECT_ORION_SERVICES_LATE_RUNS_ENABLED` → `PREFECT_API_SERVICES_LATE_RUNS_ENABLED`
    - `PREFECT_ORION_SERVICES_FLOW_RUN_NOTIFICATIONS_ENABLED` → `PREFECT_API_SERVICES_FLOW_RUN_NOTIFICATIONS_ENABLED`
    - `PREFECT_ORION_SERVICES_PAUSE_EXPIRATIONS_ENABLED` → `PREFECT_API_SERVICES_PAUSE_EXPIRATIONS_ENABLED`
    - `PREFECT_ORION_TASK_CACHE_KEY_MAX_LENGTH` → `PREFECT_API_TASK_CACHE_KEY_MAX_LENGTH`
    - `PREFECT_ORION_SERVICES_CANCELLATION_CLEANUP_ENABLED` → `PREFECT_API_SERVICES_CANCELLATION_CLEANUP_ENABLED`

### CODE REVIEW

File: docs/concepts/deployments.md [hunk 1]
```diff
@@ -353,7 +352,7 @@ When you run a deployed flow with Prefect Orion, the following happens:
 [Agents and work pools](/concepts/work-pools/) enable the Prefect orchestration engine and API to run deployments in your local execution environments. To execute deployed flow runs you need to configure at least one agent.
 
 !!! note "Scheduled flow runs"
-    Scheduled flow runs will not be created unless the scheduler is running with either Prefect Cloud or a local Prefect Orion API server started with `prefect orion start`.
+    Scheduled flow runs will not be created unless the scheduler is running with either Prefect Cloud or a local Prefect server started with `prefect orion start`.
```
Comment (zhen0) [inline docs/concepts/deployments.md:355]: Should this be 'prefect server start'?

File: docs/concepts/deployments.md [hunk 2]
```diff
@@ -486,7 +485,8 @@ $ prefect deployment inspect 'Cat Facts/catfact'
 ## Create a flow run from a deployment
 
 ### Create a flow run with a schedule
-If you specify a schedule for a deployment, the deployment will execute its flow automatically on that schedule as long as a Prefect Orion API server and agent is running. Prefect Cloud created scheduled flow runs automatically, and they will run on schedule if an agent is configured to pick up flow runs for the deployment.
+
+If you specify a schedule for a deployment, the deployment will execute its flow automatically on that schedule as long as a Prefect server and agent is running. Prefect Cloud-created schedules flow runs automatically, and they will run on schedule if an agent is configured to pick up flow runs for the deployment.
```
Comment (zhen0) [inline docs/concepts/deployments.md:489]: Tiny typo - Prefect Cloud created....

File: docs/concepts/settings.md [hunk 3]
```diff
@@ -39,11 +39,11 @@ PREFECT_API_URL="https://api.prefect.cloud/api/accounts/[ACCOUNT-ID]/workspaces/
...
+!!! tip "Running the Prefect UI behind a reverse proxy"
+    When using a reverse proxy (such as [Nginx](https://nginx.org) or [Traefik](https://traefik.io)) to proxy traffic to a locally-hosted Prefect UI instance, the Prefect server also needs to be configured to know how to connect to the API. The  [`PREFECT_ORION_UI_API_URL`](prefect/settings/#prefect.settings.PREFECT_ORION_UI_API_URL)  should be set to the external proxy URL (e.g. if your external URL is https://prefect-orion.example.com/ then set `PREFECT_ORION_UI_API_URL=https://prefect-orion.example.com/api` for the Prefect server process).  You can also accomplish this by setting [`PREFECT_API_URL`](/concepts/settings/#prefect.settings.PREFECT_API_URL) to the API URL, as this setting is used as a fallback if `PREFECT_ORION_UI_API_URL` is not set.
```
Comment (zhen0) [inline docs/concepts/settings.md:46]: Do we want to keep this as PREFECT_ORION_UI_API_URL ?

File: docs/faq.md [hunk 4]
```diff
@@ -103,13 +95,13 @@ See the [Installation](/getting-started/installation/) documentation and [Window
...
+Prefect Cloud users do not need to worry about the Prefect database. Prefect Cloud uses PostgreSQL on GCP behind the scenes. To use PostgreSQL with a self-hosted Prefect server, users must provide the [connection string][prefect.settings.PREFECT_ORION_DATABASE_CONNECTION_URL] for a running database via the `PREFECT_ORION_DATABASE_CONNECTION_URL` environment variable.
 
 ### What databases does Prefect support?
 
-A self-hosted Prefect Orion server can work with SQLite and PostgreSQL. New Prefect installs default to a SQLite database hosted at `~/.prefect/orion.db` on Mac or Linux machines. SQLite and PostgreSQL are not installed by Prefect.
+A self-hosted Prefect server can work with SQLite and PostgreSQL. New Prefect installs default to a SQLite database hosted at `~/.prefect/orion.db` on Mac or Linux machines. SQLite and PostgreSQL are not installed by Prefect.
```
Comment (zhen0) [inline docs/faq.md:104]: Do we want to remove the Orion ref here?

File: docs/api-ref/overview.md [hunk 5]
```diff
@@ -15,11 +15,11 @@ Prefect 2 provides a number of programmatic workflow interfaces. Each API is doc
...
+- The [Prefect REST API](/api-ref/rest-api/) is used for communicating data from clients to the Prefect server so that orchestration can be performed. This API is mainly consumed by clients like the Prefect Python Client or the server dashboard.
 
 !!! note "Prefect REST API interactive documentation"
     Prefect Cloud REST API documentation is available at <a href="https://app.prefect.cloud/api/docs" target="_blank">https://app.prefect.cloud/api/docs</a>.
 
-    The Prefect Orion REST API documentation for a local instance run with with `prefect orion start` is available at <a href="http://localhost:4200/docs" target="_blank">http://localhost:4200/docs</a> or the `/docs` endpoint of the [`PREFECT_API_URL`](/concepts/settings/#prefect_api_url) you have configured to access the server.
+    The Prefect REST API documentation for a local instance run with with `prefect orion start` is available at <a href="http://localhost:4200/docs" target="_blank">http://localhost:4200/docs</a> or the `/docs` endpoint of the [`PREFECT_API_URL`](/concepts/settings/#prefect_api_url) you have configured to access the server.
```
Comment (peytonrunyan) [inline docs/api-ref/overview.md:24]: Should this still be `prefect orion start`?

File: docs/concepts/settings.md (hunk 3 above)
Comment (madkinsz) [reply docs/concepts/settings.md:46]: That'll be updated in a pull request today.

File: docs/faq.md (hunk 4 above)
Comment (madkinsz) [reply docs/faq.md:104]: Thanks!

File: docs/api-ref/overview.md (hunk 5 above)
Comment (madkinsz) [reply docs/api-ref/overview.md:24]: Will fix, thanks!

File: docs/api-ref/overview.md [hunk 6]
```diff
@@ -15,11 +15,11 @@ Prefect 2 provides a number of programmatic workflow interfaces. Each API is doc
 
 - The Prefect Python API is used to build, test, and execute workflows against the Orion workflow engine. This is the primary user-facing API.
 - The Prefect Orion Python API is used by the Orion server to work with workflow metadata and enforce orchestration logic. This API is primarily used by Orion developers.
```
Comment (peytonrunyan) [inline docs/api-ref/overview.md:17]: Should these two bullets still reference Orion ?

**Review by znicholasbrown — APPROVED** [review]

**Pull request merged**